No existing entry for trainer_params.mode.type
{}[type] = <class 'int'>(1)
No existing entry for trainer_params.mode.objective_type
{'type': 1}[objective_type] = <class 'str'>(normal)
No existing entry for trainer_params.mode.use_marginal
{'type': 1, 'objective_type': 'normal'}[use_marginal] = <class 'bool'>(True)
No existing entry for trainer_params.mode.nb_samples
{'type': 1, 'objective_type': 'normal', 'use_marginal': True}[nb_samples] = <class 'int'>(10)
No existing entry for trainer_params.mode.temperature
{'type': 1, 'objective_type': 'normal', 'use_marginal': True, 'nb_samples': 10}[temperature] = <class 'float'>(5.0)
No existing entry for trainer_params.mode.scale_temperature_by_path_length
{'type': 1, 'objective_type': 'normal', 'use_marginal': True, 'nb_samples': 10, 'temperature': 5.0}[scale_temperature_by_path_length] = <class 'bool'>(True)
No existing entry for trainer_params.mode.use_gamma
{'type': 1, 'objective_type': 'normal', 'use_marginal': True, 'nb_samples': 10, 'temperature': 5.0, 'scale_temperature_by_path_length': True}[use_gamma] = <class 'bool'>(True)
No existing entry for trainer_params.mode.loss_type
{'type': 1, 'objective_type': 'normal', 'use_marginal': True, 'nb_samples': 10, 'temperature': 5.0, 'scale_temperature_by_path_length': True, 'use_gamma': True}[loss_type] = <class 'str'>(mse)
No existing entry for trainer_params.mode.perturb_w
{'type': 1, 'objective_type': 'normal', 'use_marginal': True, 'nb_samples': 10, 'temperature': 5.0, 'scale_temperature_by_path_length': True, 'use_gamma': True, 'loss_type': 'mse'}[perturb_w] = <class 'bool'>(True)
No existing entry for trainer_params.mode.perturb_w_prime
{'type': 1, 'objective_type': 'normal', 'use_marginal': True, 'nb_samples': 10, 'temperature': 5.0, 'scale_temperature_by_path_length': True, 'use_gamma': True, 'loss_type': 'mse', 'perturb_w': True}[perturb_w_prime] = <class 'bool'>(False)
No existing entry for trainer_params.mode.type
{}[type] = <class 'int'>(1)
No existing entry for trainer_params.mode.objective_type
{'type': 1}[objective_type] = <class 'str'>(normal)
No existing entry for trainer_params.mode.use_marginal
{'type': 1, 'objective_type': 'normal'}[use_marginal] = <class 'bool'>(True)
No existing entry for trainer_params.mode.nb_samples
{'type': 1, 'objective_type': 'normal', 'use_marginal': True}[nb_samples] = <class 'int'>(10)
No existing entry for trainer_params.mode.temperature
{'type': 1, 'objective_type': 'normal', 'use_marginal': True, 'nb_samples': 10}[temperature] = <class 'float'>(5.0)
No existing entry for trainer_params.mode.scale_temperature_by_path_length
{'type': 1, 'objective_type': 'normal', 'use_marginal': True, 'nb_samples': 10, 'temperature': 5.0}[scale_temperature_by_path_length] = <class 'bool'>(True)
No existing entry for trainer_params.mode.use_gamma
{'type': 1, 'objective_type': 'normal', 'use_marginal': True, 'nb_samples': 10, 'temperature': 5.0, 'scale_temperature_by_path_length': True}[use_gamma] = <class 'bool'>(True)
No existing entry for trainer_params.mode.loss_type
{'type': 1, 'objective_type': 'normal', 'use_marginal': True, 'nb_samples': 10, 'temperature': 5.0, 'scale_temperature_by_path_length': True, 'use_gamma': True}[loss_type] = <class 'str'>(mse)
No existing entry for trainer_params.mode.perturb_w
{'type': 1, 'objective_type': 'normal', 'use_marginal': True, 'nb_samples': 10, 'temperature': 5.0, 'scale_temperature_by_path_length': True, 'use_gamma': True, 'loss_type': 'mse'}[perturb_w] = <class 'bool'>(True)
No existing entry for trainer_params.mode.perturb_w_prime
{'type': 1, 'objective_type': 'normal', 'use_marginal': True, 'nb_samples': 10, 'temperature': 5.0, 'scale_temperature_by_path_length': True, 'use_gamma': True, 'loss_type': 'mse', 'perturb_w': True}[perturb_w_prime] = <class 'bool'>(False)
{
    "evaluate_every": 5,
    "loader_params": {
        "data_dir": "data/warcraft_shortest_path/12x12",
        "evaluate_with_extra": false,
        "normalize": true,
        "use_local_path": false,
        "use_test_set": true
    },
    "model_dir": "results/warcraft_shortest_path_combresnet",
    "num_cpus": 20,
    "num_epochs": 50,
    "problem_type": "warcraft_shortest_path",
    "ray_params": {},
    "save_visualizations": false,
    "seed": 1,
    "trainer_name": "DijkstraMAP",
    "trainer_params": {
        "batch_size": 70,
        "l1_regconst": 0.0,
        "lambda_val": 20.0,
        "lr_milestone_1": 30,
        "lr_milestone_2": 40,
        "mode": {
            "loss_type": "mse",
            "nb_samples": 10,
            "objective_type": "normal",
            "perturb_w": true,
            "perturb_w_prime": false,
            "scale_temperature_by_path_length": true,
            "temperature": 5.0,
            "type": 1,
            "use_gamma": true,
            "use_marginal": true
        },
        "model_params": {
            "arch_params": {},
            "model_name": "CombResnet18"
        },
        "neighbourhood_fn": "8-grid",
        "optimizer_name": "Adam",
        "optimizer_params": {
            "lr": 0.0005
        },
        "preload_batch": true,
        "use_cuda": true,
        "use_lr_scheduling": true
    },
    "use_ray": true
}
Traceback (most recent call last):
  File "./cli/warcraft-cli.py", line 176, in <module>
    main()
  File "./cli/warcraft-cli.py", line 133, in main
    **params.get("ray_params", {})
  File "/home/pminervi/anaconda3/envs/gpu/lib/python3.6/site-packages/ray/worker.py", line 717, in init
    ray_params=ray_params)
  File "/home/pminervi/anaconda3/envs/gpu/lib/python3.6/site-packages/ray/node.py", line 200, in __init__
    self.start_head_processes()
  File "/home/pminervi/anaconda3/envs/gpu/lib/python3.6/site-packages/ray/node.py", line 805, in start_head_processes
    self.start_redis()
  File "/home/pminervi/anaconda3/envs/gpu/lib/python3.6/site-packages/ray/node.py", line 580, in start_redis
    fate_share=self.kernel_fate_share)
  File "/home/pminervi/anaconda3/envs/gpu/lib/python3.6/site-packages/ray/_private/services.py", line 716, in start_redis
    fate_share=fate_share)
  File "/home/pminervi/anaconda3/envs/gpu/lib/python3.6/site-packages/ray/_private/services.py", line 862, in _start_redis_instance
    wait_for_redis_to_start("127.0.0.1", port, password=password)
  File "/home/pminervi/anaconda3/envs/gpu/lib/python3.6/site-packages/ray/_private/services.py", line 546, in wait_for_redis_to_start
    raise RuntimeError("Unable to connect to Redis. If the Redis instance "
RuntimeError: Unable to connect to Redis. If the Redis instance is on a different machine, check that your firewall is configured properly.
